aiida_workgraph/web/frontend/src/components/DataNodeTable.js

```diff
@@ -68,6 +69,51 @@ function DataNode() {
         .catch(error => console.error('Error deleting item: ', error));
     };
 
+    const [toDeleteItem, setToDeleteItem] = useState(null);
+    const [showConfirmDeleteModal, setShowConfirmDeleteModal] = useState(false);
+    const [bodyTextConfirmDeleteModal, setBodyTextConfirmDeleteModal] =  useState(<p></p>);
+    // useEffect to ensure this happens after the toDeletItem has been updated by the delete button click
+    useEffect(() => {
+        if (toDeleteItem != null) {
+          fetch(`http://localhost:8000/api/datanode/delete/${toDeleteItem.pk}?dry_run=True`, {
+              method: 'DELETE',
+          })
+          .then(response => response.json())
+          .then(data => {
+              if (data.deleted_nodes.length > 0) {
+                // the node {item.pk} will be always in the list if the deletion is successfull
+                if (data.deleted_nodes.length > 1) {
+                  let formatted_pks = data.deleted_nodes.map((x) => ` PK<${x.toString()}>`);
+                  data.deleted_nodes.splice(data.deleted_nodes.indexOf(toDeleteItem.pk), 1)
+                  setBodyTextConfirmDeleteModal(
+                    <p>
+                    Are you sure you want to delete node PK&lt;{toDeleteItem.pk}&gt; and {data.deleted_nodes.length} its dependencies?
```

Comment:
Use `dependent nodes` is more clear to the user.
```suggestion
                    Are you sure you want to delete node PK&lt;{toDeleteItem.pk}&gt; and {data.deleted_nodes.length} its dependent nodes.?
```